e2e/pages/enterOtherProceedings/enterOtherProceedings.js

```diff
@@ -1,73 +1,75 @@
 const I = actor();
 
 module.exports = {
-  state: {
-    contextPrefix: '',
-  },
-
-  fields: function() {
-    const prefix = this.state.contextPrefix;
-
+  fields: function(index) {
     return {
       onGoingProceeding: {
         yes: '#proceeding_onGoingProceeding-Yes',
         no: '#proceeding_onGoingProceeding-No',
         dontKnow: '#proceeding_onGoingProceeding-DontKnow',
       },
       proceedingStatus: {
-        previous: `#proceeding_${prefix}proceedingStatus-Previous`,
-        ongoing: `#proceeding_${prefix}proceedingStatus-Ongoing`,
+        previous: `#proceeding_${index}proceedingStatus-Previous`,
+        ongoing: `#proceeding_${index}proceedingStatus-Ongoing`,
       },
-      caseNumber: `#proceeding_${prefix}caseNumber`,
-      started: `#proceeding_${prefix}started`,
-      ended: `#proceeding_${prefix}ended`,
-      ordersMade: `#proceeding_${prefix}ordersMade`,
-      judge: `#proceeding_${prefix}judge`,
-      children: `#proceeding_${prefix}children`,
-      guardian: `#proceeding_${prefix}guardian`,
+      caseNumber: `#proceeding_${index}caseNumber`,
+      started: `#proceeding_${index}started`,
+      ended: `#proceeding_${index}ended`,
+      ordersMade: `#proceeding_${index}ordersMade`,
+      judge: `#proceeding_${index}judge`,
+      children: `#proceeding_${index}children`,
+      guardian: `#proceeding_${index}guardian`,
       sameGuardianNeeded: {
-        yes: `#proceeding_${prefix}sameGuardianNeeded-Yes`,
-        no: `#proceeding_${prefix}sameGuardianNeeded-No`,
+        yes: `#proceeding_${index}sameGuardianNeeded-Yes`,
+        no: `#proceeding_${index}sameGuardianNeeded-No`,
       },
     };
   },
 
   addProceedingButton: 'Add new',
 
   addNewProceeding() {
-    if (this.state.contextPrefix === 'additionalProceedings_0_') {
-      throw new Error('Adding more proceedings is not supported in the test');
-    }
-
     I.click(this.addProceedingButton);
-    this.state.contextPrefix = 'additionalProceedings_0_';
   },
 
   selectYesForProceeding() {
-    I.click(this.fields().onGoingProceeding.yes);
+    I.click(this.fields(undefined).onGoingProceeding.yes);
```

Comment:
Do we need to pass the undefined explicitly? It would be undefined if we don't pass anything, wouldn't it?

Reply:
You are right, I just preferred to be explicit.